src/main/java/org/isf/menu/service/MenuIoOperations.java

```diff
@@ -313,30 +324,27 @@ private GroupMenu insertGroupMenu(UserGroup aGroup, UserMenuItem item) throws OH
 
 	/**
 	 * Deletes a {@link UserGroup}
-	 * 
 	 * @param aGroup - the {@link UserGroup} to delete
 	 * @throws OHServiceException When failed to delete group
 	 */
 	public void deleteGroup(UserGroup aGroup) throws OHServiceException {
-		groupMenuRepository.deleteWhereUserGroup(aGroup.getCode());
-		groupPermissionIoOperationRepository.deleteAllByUserGroup_Code(aGroup.getCode());
+		ensureUserGroupNotDeleted(aGroup.getCode());
 		groupRepository.delete(aGroup);
 	}
 
 	/**
 	 * Insert a new {@link UserGroup} with a minimum set of rights
-	 * 
 	 * @param aGroup - the {@link UserGroup} to insert
 	 * @return the new {@link UserGroup}
 	 * @throws OHServiceException When failed to create group
 	 */
 	public UserGroup newUserGroup(UserGroup aGroup) throws OHServiceException {
+		aGroup.setDeleted(false);
 		return groupRepository.save(aGroup);
 	}
```

Comment:
You are suggesting that every time we have an entity with a `_DELETED` field we have to be sure that `newEntity...` methods set that field to `false`?

Reply:
Yes !

Reply:
I don't think it worth it... when you create new entities, booleans are false by default, and your suggestion is just adding a bug that will be also difficult to spot when an admin wants to create an object directly in deleted status (e.g. for tests)

Reply:
Let me remove it then !